kornia/core/module.py

```diff
@@ -11,6 +13,51 @@
 from .external import numpy as np
 
 
+class ONNXExportMixin:
+    ONNX_EXPORTABLE: bool = True
+    ONNX_DEFAULT_INPUTSHAPE: ClassVar[list[int]] = [-1, -1, -1, -1]
+    ONNX_DEFAULT_OUTPUTSHAPE: ClassVar[list[int]] = [-1, -1, -1, -1]
+
+    def to_onnx(
+        self,
+        onnx_name: Optional[str] = None,
+        input_shape: Optional[list[int]] = None,
+        output_shape: Optional[list[int]] = None,
+    ) -> None:
+        if not self.ONNX_EXPORTABLE:
+            raise RuntimeError("This object cannot be exported to ONNX.")
+
+        if input_shape is None:
+            input_shape = self.ONNX_DEFAULT_INPUTSHAPE
+        if output_shape is None:
+            output_shape = self.ONNX_DEFAULT_OUTPUTSHAPE
+
+        if onnx_name is None:
+            onnx_name = f"Kornia-{self.__class__.__name__}.onnx"
```

Comment:
```suggestion
            onnx_name = f"kornia-{kornia.__verision__}-{self.__class__.__name__}.onnx"
```
or add some sort of metadata, maybe as discussed by adding it to the proto metadata fields
https://qgis-plugin-deepness.readthedocs.io/en/latest/creators/creators_add_metadata_to_model.html